### PR TITLE
Dstu2 export

### PR DESCRIPTION
**Pull request by kjmahalingam**
This PR implements DSTU2 export of patients. The DSTU2 exporter is based on the STU3 exporter, with certain changes, removals, and additions of fields in order to produce valid DSTU2 FHIR.

Additionally, this PR implements:
- DSTU2 FHIR export validation testing
- DSTU2 FHIR hospital export
- Updated HAPI FHIR libraries (3.0.0 to 3.1.0)
- More descriptive gradle test failure output

### CODE REVIEW

File: src/main/java/org/mitre/synthea/export/FhirDstu2.java [hunk 1]
```diff
@@ -0,0 +1,1233 @@
...
+    if (person.attributes.get(Person.IDENTIFIER_DRIVERS) != null) {
+      patientResource.addIdentifier()
+          .setType(IdentifierTypeCodesEnum.DL)
+          .setSystem("urn:oid:2.16.840.1.113883.4.3.25")
+          .setValue((String) person.attributes.get(Person.IDENTIFIER_DRIVERS));
+    }
+
+    // We do not yet account for mixed race
+    ExtensionDt raceExtension = new ExtensionDt();
+    raceExtension.setUrl("http://hl7.org/fhir/us/core/StructureDefinition/us-core-race");
```
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/export/FhirDstu2.java]: In DSTU, the extension is `http://hl7.org/fhir/StructureDefinition/us-core-race`
See http://hl7.org/fhir/DSTU2/extension-us-core-race.html

File: src/main/java/org/mitre/synthea/export/FhirDstu2.java [hunk 2]
```diff
@@ -0,0 +1,1233 @@
...
+    raceTextExtension.setUrl("text");
+    raceTextExtension.setValue(new StringDt(raceDisplay));
+
+    raceExtension.addUndeclaredExtension(raceTextExtension);
+
+    patientResource.addUndeclaredExtension(raceExtension);
+
+    // We do not yet account for mixed ethnicity
+    ExtensionDt ethnicityExtension = new ExtensionDt();
+    ethnicityExtension.setUrl("http://hl7.org/fhir/us/core/StructureDefinition/us-core-ethnicity");
```
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/export/FhirDstu2.java]: In DSTU2, the extension is `http://hl7.org/fhir/StructureDefinition/us-core-ethnicity`
See http://hl7.org/fhir/DSTU2/extension-us-core-ethnicity.html

Comment (jawalonoski) [reply src/main/java/org/mitre/synthea/export/FhirDstu2.java]: The structure is also different than the STU3 structure.

File: src/main/java/org/mitre/synthea/export/FhirDstu2.java (hunk 1 above)
Comment (jawalonoski) [reply src/main/java/org/mitre/synthea/export/FhirDstu2.java]: The structure is also different than the STU3 structure.

File: src/main/java/org/mitre/synthea/export/FhirDstu2.java [hunk 3]
```diff
@@ -0,0 +1,1233 @@
...
+      if (person.attributes.get(Person.NAME_PREFIX) != null) {
+        maidenName.addPrefix((String) person.attributes.get(Person.NAME_PREFIX));
+      }
+      if (person.attributes.get(Person.NAME_SUFFIX) != null) {
+        maidenName.addSuffix((String) person.attributes.get(Person.NAME_SUFFIX));
+      }
+    }
+
+    ExtensionDt birthSexExtension = new ExtensionDt();
+    birthSexExtension.setUrl("http://hl7.org/fhir/us/core/StructureDefinition/us-core-birthsex");
```
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/export/FhirDstu2.java]: This extension does not exist in DSTU2.

**Comment by kjmahalingam** [comment]
Patient extensions have been fixed, as discussed.

**Pull request merged**